### DISCUSSION TITLE
clarification on lighting

### DISCUSSION THREAD

**Discussion by linlumio** [sceneview/sceneview#319] [Q&A]
Hi! I'm programming an application to display indicative arrows. The three-dimensional GLB models are:

https://drive.google.com/file/d/1M897BD1hupHYqVYC4FUOduUtowwg_-af/view?usp=sharing, https://drive.google.com/file/d/1UP2XRAvzrCUYRizU7_AaoGmD1h4cxiPc/view?usp=sharing

_first doubt:_ Until now I have been using sceneview 0.9.4, I don't need the lighting management, because the arrows should be an indication always visible in any situation. But every time I placed the 3D object I saw it very dark (white arrow becomes black).
I tried modifying:
```
arSceneView.mainLight
arSceneView.scene.indirectLight
arSceneView.environment.indirectLight
arSceneView.addLight()
arModelNode.setCastShadows(true / false)
arModelNode.setReceiveShadows(true / false)
```

and setting `arSession.lightEstimationMode = Config.LightEstimationMode.DISABLED`

Eventually setting `arSession.lightEstimationMode = Config.LightEstimationMode.ENVIRONMENTAL_HDR`
the colours looked good, but the system was still handling the lighting, which I did not want. Is there any way to retain the original colours of a 3D object without them being affected by the lighting?

_Second doubt:_ Updating to sceneview 0.10.2 brought back the problems described above: 3D objects are very dark and this time changing 'LightEstimationMode' seems to have no effect.
If I place the objects close together, I can add a light:
```
arSceneView.addLight(
                LightManager.Builder(LightManager.Type.POINT)
                    .apply {
                        val (r,g,b) = Colors.cct(4_000f)
                        color(r,g,b)
                        intensity(2_000_000f)
                        castShadows(true)
                    }.build()
            )
```

but as I place an object some distance away it is shown dark.
Any suggestion?

p.s. Thank you very much for your incredible work with this library!

**Comment by sameerjj** [comment]
I have pretty much the same issue! I also added an IndirectLight
```
sceneView.indirectLight = IndirectLight.Builder()
    .intensity(100000f)
    .build(sceneView.engine)
```
but no satisfying results.

**Comment by tomriddle25** [comment]
Hi folks, I am in a similar situation, did you find an answer to this ?

**Comment by ThomasGorisse** [comment]
@sameerjj Could you create a PR with your indirect by default?

**Reply by sameerjj** [reply]
Hi @ThomasGorisse wasn't 100% sure what you were requesting but I submitted my attempt:
https://github.com/SceneView/sceneview-android/pull/460

**Comment by ThomasGorisse** [comment]
In v4.0.0-rc.1, lighting is handled via environment (`.hdr` indirect light) and `LightNode` (direct lights):

```kotlin
SceneView {
    // Environment (IBL) is set via SceneView parameters
    // Direct light:
    LightNode(
        apply = {
            intensity(100_000f)
            color(1f, 1f, 1f)
            direction(0f, -1f, 0f)
        }
    )
    ModelNode(modelInstance = model)
}
```

**Important**: `LightNode`'s `apply` is a **named parameter** (`apply = { ... }`), not a trailing lambda. The environment HDR provides ambient/indirect lighting, while `LightNode` adds directional, point, or spot lights.
